File: frontend/angular-book-ecommerce/src/app/common/address.ts
export class Address {
    street: string;
    city: string;
    district: string;
    country: string;
    zipCode: string;
    specificAddress: string;

    constructor(street: string, city: string, district: string, country: string, zipCode: string, specificAddress: string) {
        this.street = street;
        this.city = city;
        this.district = district;
        this.country = country;
        this.zipCode = zipCode;
        this.specificAddress = specificAddress;
    }

}

File: backend/spring-boot-book-ecommerce/src/main/java/com/luvannie/springbootbookecommerce/entity/Address.java
package com.luvannie.springbootbookecommerce.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "address")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Address {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "street")
    private String street;

    @Column(name = "city")
    private String city;

    @Column(name = "district")
    private String district;

    @Column(name = "country")
    private String country;

    @Column(name = "zip_code")
    private String zipCode;

    @Column(name = "specific_address")
    private String specificAddress;

    @OneToOne
    @PrimaryKeyJoinColumn
    private Order order;
}
